rdrf/rdrf/management/commands/check_structure.py

```diff
@@ -0,0 +1,105 @@
+import sys
+from django.core.management import BaseCommand
+from rdrf.models import Registry
+from rdrf.models import Modjgo
+import yaml
+import jsonschema
+import errno
+import os
+
+explanation = "This command checks for schema validation errors"
+
+SCHEMA_LOCATIONS = ["/app/rdrf/rdrf/schemas/modjgo.yaml",
+                    "/env/src/django-rdrf/rdrf/rdrf/schemas/modjgo.yaml"]
+
+class Command(BaseCommand):
+    help = 'Checks in clinical db against json schema(s)'
+
+    def add_arguments(self, parser):
+        parser.add_argument('-r',"--registry_code",
+                            action='store',
+                            dest='registry_code',
+                            help='Code of registry to check')
+        parser.add_argument('-c','--collection',
+                            action='store',
+                            dest='collection',
+                            default="cdes",
+                            choices=['cdes', 'history', 'progress', 'registry_specific'],
+                            help='Collection name')
+
+        parser.add_argument('-t','--test-mode',
+                            action='store_true',
+                            dest='test_mode',
+                            default=False,
+                            help='Test mode - does not return exit code 1 on fail')
+
+    def _usage(self):
+        print(explanation)
+
+    def _print(self, msg):
+        self.stdout.write(msg + "\n")
+
+    def handle(self, *args, **options):
+        self.test_mode = options.get("test_mode", False)
+        problem_count = 0
+        self.schema = self._load_schema()
+        registry_code = options.get("registry_code",None)
+        if registry_code is None:
+            self._print("Error: registry code required")
+            if not self.test_mode:
+                sys.exit(1)
+        try:
+            registry_model = Registry.objects.get(code=registry_code)
+        except Registry.DoesNotExist:
+            self._print("Error: registry does not exist")
+            if not self.test_mode:
+                sys.exit(1)
```

Comment:
Actually I did this on purpose - I'll have to remember why now   ,initially I tested for /expected sys exit in the tests, i'll have to re-try it  to prod my  memory